### ISSUE TITLE
[ember/no-side-effects] False-positive with an assignment in a lambda

### ISSUE THREAD

**Issue by boris-petrov** [ember-cli/eslint-plugin-ember#1156]
```typescript
public get foo(): object {
  return {
    bar: x => {
      this.x = x;
    },
  };
}
```

This leads to a `ember/no-side-effects` which is wrong.

**Comment by bmish** [comment]
We had some previous discussions about this case in #915 and #917. At the time, we decided not to make any changes to the rule, because closures/lambdas did not seem like a reliable heuristic.

**Comment by boris-petrov** [comment]
Ah, sorry, I didn't see those. I'm fine if you close it as a `wontfix`. However, some support for this would be nice, as that's a pretty normal use-case I believe.

**Comment by bmish** [comment]
I'll close for now since I don't have a perfect fix for it.

**Comment by boris-petrov** [comment]
@bmish - well, it's good to have an open issue for this so it's not forgotten but it's your choice. :)
